File: benchmarks/mandelbrot.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
total = 0
y = -1.5
while y < 1.5
  x = -2.0
  while x < 1.0
    zr = 0.0
    zi = 0.0
    i = 0
    while i < 255 && zr * zr + zi * zi <= 4.0
      nzr = zr * zr - zi * zi + x
      nzi = 2.0 * zr * zi + y
      zr = nzr
      zi = nzi
      i += 1
    end
    total += i
    x += 0.005
  end
  y += 0.005
end
t1 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)

puts total
puts "bench_ns=#{t1 - t0}"

File: benchmarks/mandelbrot.js
const t0 = process.hrtime.bigint();
let total = 0;
let y = -1.5;
while (y < 1.5) {
    let x = -2.0;
    while (x < 1.0) {
        let zr = 0.0;
        let zi = 0.0;
        let i = 0;
        while (i < 255 && zr * zr + zi * zi <= 4.0) {
            let nzr = zr * zr - zi * zi + x;
            let nzi = 2.0 * zr * zi + y;
            zr = nzr;
            zi = nzi;
            i += 1;
        }
        total += i;
        x += 0.005;
    }
    y += 0.005;
}
const t1 = process.hrtime.bigint();

console.log(total);
console.log(`bench_ns=${t1 - t0}`);
